circleci.sh

```diff
@@ -52,18 +52,19 @@ posturl="https://circleci.com/api/v1.1/project/github/$githubAccountName/$projec
 
 echo "Loading environment variables into project $githubAccountName/$projectName:"
 
+response="";
 for variable in $variables;
 do
     
     set -- `echo $variable | tr '=' ' '`
-    curl -X POST --header "Content-Type: application/json" -d "{\"name\":\"$1\", \"value\":\"$2\"}" $posturl
-
+    response+=$(curl -X POST --header "Content-Type: application/json" -d "{\"name\":\"$1\", \"value\":\"$2\"}" $posturl)
```

Comment:
This could do with some fail-fast/short-circuit. What about checking for the target URL only once before the for-loop?

Reply:
At the moment we don't check if every (or any) call is successful, and that's why I felt the need to save the response of each call and then query it to verify that we really set the environment variables.
So far it could fail for many reasons, not only the target url, but also for a misspelled repository, empty values, wrong username and so on... I see your point of failing fast, but I think it would come with the cost of complexity (how many scenarios do we have, how do we cover them in a script, etc...) so I opted for this approach instead. 

Hope this makes sense?

Reply:
Sure. But what about just getting the if inside of the loop, therefore stopping the process as soon as there is an error, which is inline with the bash options being used: set -euo pipefail

Also, that would block in between states such as the below:

Variable1=value (OK)
Variable2=value (Failed)
Variable3=value (OK)
Variable4=value (Failed)
etc...

Short-circuiting would cancel the process before triggering the 3rd request, making it easy to debug problems as they come along.

Now if you are very passioned about allowing the process to continue upon errors, it would make sense to change the entire script accordingly and then summarise all the issues at the bottom. What do you reckon?

Reply:
That actually makes completely sense! HELL YEAH!
Will have a look shortly, I'm finally going back to this project after a few months of circle-ci abstinence ... @pjbgf 😅